cmd/kubeadm/app/util/config/cluster.go

```diff
@@ -180,25 +183,72 @@ func getNodeNameFromKubeletConfig(kubeconfigDir string) (string, error) {
 	return strings.TrimPrefix(cert.Subject.CommonName, constants.NodesUserPrefix), nil
 }
 
+// getAPIEndpointOrMigrate returns the APIEndpoint for the current node as reported by getAPIEndpoint. If this
+// information is not yet present, chances are that we haven't yet migrated this information to reside on
+// pod annotations, fallback in that case to GetClusterStatus, that will compose this information out of the
+// ClusterStatus struct present in the kubeadm-config ConfigMap.
+//
+// TODO: remove after 1.20, when the ClusterStatus struct is removed from the kubeadm-config ConfigMap;
+//       callers of this function should call getAPIEndpoint directly when that happens.
+func getAPIEndpointOrMigrate(client clientset.Interface, nodeName string, apiEndpoint *kubeadmapi.APIEndpoint) error {
```

Comment:
This name is a bit misleading. We don't do any migration here.
I would suggest to keep the `getAPIEndpoint` name here. Then make a couple of helpers that have an implementation specific name in their name - `getAPIEndpointFromPod` and `getAPIEndpointFromClusterStatus`. The top level func (`getAPIEndpoint`) would call just the helpers then.

Reply:
Yes, you're right, I also found the name a bit funky. Will update.